plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -388,9 +390,12 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
         String targetPath = getTableLocation(tableMetadata.getProperties())
                 .orElseGet(() -> catalog.defaultTableLocation(session, schemaTableName));
 
+        Boolean isHiveEnabled = getHiveEnabledFlag(tableMetadata.getProperties()).orElse(false);
+
         ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builderWithExpectedSize(2);
         FileFormat fileFormat = IcebergTableProperties.getFileFormat(tableMetadata.getProperties());
         propertiesBuilder.put(DEFAULT_FILE_FORMAT, fileFormat.toString());
+        propertiesBuilder.put(HIVE_ENABLED, isHiveEnabled.toString());
```

Comment:
Thats right, but whats the other approach you are thinking of ? I didnt find btter solution to make it happen on table level ?

Reply:
Thats the similar approach whats is done on spark side when iceberg jar is added.

Reply:
> Thats right, but whats the other approach you are thinking of ?

Add a boolean entry in some Iceberg config class.
This should be a new config class bound in `IcebergHiveMetastoreCatalogModule`, because it's going to be HMS-specific. See `IcebergConfig` & `TestIcebergConfig` for how to create a config class.
See https://github.com/trinodb/trino/pull/10845 for upcoming Glue support, which won't need this.

Then, pass the config value to `HiveMetastoreTableOperationsProvider` > `HiveMetastoreTableOperations` > `AbstractMetastoreTableOperations` and use there.